iceoryx_examples/icecubetray/README.md

```diff
@@ -0,0 +1,127 @@
+                                 | w group: infotainment |           |           #         |
+                                 |                       |           |          #          |
+                                 |                       |           |     #   #           |
+                                 |                       |           |      # #            |
+                                 |                       |           +---------------------+
+                                 +-----------------------+
+```
+
+### RouDi and Apps
+
+#### Working setup
+
+RouDi is built with two static shared memory segments _infotainment_ and _privileged_. The access rights of the segments are configured as depicted in the graphic above.
+
+The `roudiConfig` is composed of a memory pool config called `mepooConfig`. When the segement is created, one needs to
+specific the reader group (first string), writer group (second string) as well as the `mepooConfig` (last parameter).
+
+```cpp
+iox::RouDiConfig_t roudiConfig;
+
+// Create Mempool Config
+iox::mepoo::MePooConfig mepooConfig;
+
+// We only send very small data, just one mempool per segment
+mepooConfig.addMemPool({128, 1000});
+
+/// Create an Entry for a new Shared Memory Segment from the MempoolConfig and add it to the RouDiConfig
+roudiConfig.m_sharedMemorySegments.push_back({"unprivileged", "privileged", mepooConfig});
+roudiConfig.m_sharedMemorySegments.push_back({"infotainment", "infotainment", mepooConfig});
+```
+
+The radar app is started with the user _perception_ and is sending data into the _privileged_ shared memory segment.
+
+The display app is started with the user _infotainment_. It reads the topic `{"Radar", "FrontLeft", "Object"}` from the privileged segement and forwards it as a slighty modified topic `{"Radar", "HMI-Display", "Object"}`. Because the user _infotainment_ only has write access to the infotainment segment, the data is written to this segment.
+
+!!! hint
+    It's advised to create per writer group only one shared memory segement (e.g. not two segements with `w: infotainment`).
+    In this case it wouldn't be possible to control which segment will be used.
+
+The shared memory segments can be found under `/dev/shm`
+
+```
+moss@reynholm:~$ ls -al /dev/shm
+total 60268
+drwxrwxrwt  2 root  root         100 Apr  7 19:54 .
+drwxr-xr-x  6 root  root         460 Apr  6 15:53 ..
+-rw-rw----  1 roudi iceoryx 61383328 Apr  7 19:24 iceoryx_mgmt
+-rw-rw----+ 1 roudi iceoryx   160000 Apr  7 19:24 infotainment
+-rw-rw----+ 1 roudi iceoryx   160000 Apr  7 19:24 privileged
+```
+
+!!! note
+    Note the shared memory managment segment is always available for everyone to **read** and **write**
+
+#### Not-working setup
+
+The cheeky app is started with the user _notallowed_. It has neither write nor read access to any shared memory segment. Hence, RouDi will print a warning in this case.
```

Comment:
The application will segfault whenever the application tries to write data into a memory segment to which it has no access rights (or only read access).

Reply:
That's not correct. If no write access right is found and a publisher is created the error handler is called with [`POSH__RUNTIME_NO_WRITABLE_SHM_SEGMENT`](https://github.com/eclipse-iceoryx/iceoryx/blob/abe95adc35746743351645a2864649c7eea392cd/iceoryx_posh/source/runtime/posh_runtime.cpp#L217-L221). In case a subscriber is created without read access, it currently segfaults. This will be fixed with a follow-up PR for v2.0 as this requires changes in the discovery loop.